File: C++/intersection-of-two-linked-lists.cpp
tion for singly-linked list.
 * struct ListNode {
 *  *     int val;
 *   *     ListNode *next;
 *    *     ListNode(int x) : val(x), next(NULL) {}
 *     * };
 *      */
class Solution {
private:
    int len(ListNode *head) {
        int c = 0;
        while(head) {
            ++c;
            head = head->next;
        }
        return c;
    }
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        int len_a = len(headA);
        int len_b = len(headB);
        while(len_a>len_b) {
            headA = headA->next;
            --len_a;
        }
        while(len_b>len_a) {
            headB = headB->next;
            --len_b;
        }
        while(headA!=headB){
            headA = headA->next;
            headB = headB->next;
        }
        return headA;
    }
};

File: Python/intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param two ListNodes
    # @return the intersected ListNode
    def getIntersectionNode(self, headA, headB):
        def get_len(l):
            n = 0
            while l:
                n += 1
                l = l.next
            return n
            
        a = get_len(headA)
        b = get_len(headB)
        
        while a>b:
            headA = headA.next
            a -= 1
            
        while b>a:
            headB = headB.next
            b -= 1
            
        while headA != headB:
            headA = headA.next
            headB = headB.next
            
        return headA
